sdk/core/azure-core/src/main/java/com/azure/core/http/rest/SyncRestProxy.java

```diff
@@ -208,11 +206,27 @@ static BinaryData validateLengthSync(final HttpRequest request) {
 
         final long expectedLength = Long.parseLong(request.getHeaders().getValue("Content-Length"));
 
-        long length = binaryData.getLength();
-
-        if (length > expectedLength) {
-            throw new UnexpectedLengthException(String.format(BODY_TOO_LARGE,
-                binaryData.getLength(), expectedLength), binaryData.getLength(), expectedLength);
+        Long length = binaryData.getLength();
+
+        BinaryDataContent bdc = BinaryDataHelper.getContent(binaryData);
+        if (length == null) {
+            if (bdc instanceof FluxByteBufferContent) {
+                throw new IllegalStateException("Flux Byte Buffer is not supported in Synchronous Rest Proxy.");
+            } else if (bdc instanceof InputStreamContent) {
+                LengthValidatingInputStream lengthValidatingInputStream =
+                    new LengthValidatingInputStream(((InputStreamContent) bdc).toStream(), expectedLength);
+                try {
+                    lengthValidatingInputStream.readAllBytes();
```

Comment:
we shouldn't eagerly read the stream here.  Rather substitute request body with wrapped stream and the validation will happen lazily when stream is read. See https://github.com/Azure/azure-sdk-for-java/blob/400e90a8ceced0f0ad5d586532db6637d29b45d6/sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/LengthValidatingInputStream.java#L48 .